src/client.ts

```diff
@@ -2,14 +2,147 @@ import * as Models from './models'
 import * as Mappers from './models/mappers'
 import { WebsiteAPI } from './websiteAPI'
 import { WebsiteAPIContext } from './websiteAPIContext'
+import {
+  HttpOperationResponse,
+  BaseRequestPolicy,
+  RequestPolicy,
+  RequestPolicyOptions,
+  WebResource,
+  RequestPolicyFactory,
+} from '@azure/ms-rest-js'
+
+export type ErrorType = 'ApiError' | 'BadRequest' | 'Unauthorized' | 'Forbidden' | 'NotFound' | 'InternalServerError'
+export class BaseApiError extends Error {
+  errorType: ErrorType
+  innerResponse?: HttpOperationResponse
+
+  constructor(error: Models.ApiErrorUnion, response?: HttpOperationResponse) {
+    super(error.message)
+    Object.setPrototypeOf(this, BaseApiError.prototype)
```

Comment:
Are these setPrototype needed? 
I would expect something like this to be enough:
```
class SomethingError {}
const a = new SomethingError()
console.log(a instanceof SomethingError)
```

Reply:
Needed indeed. On the same bundle you can extend an error from Error just specifying `name` to the constructor name. But for some reason it gets lost when importing the library elsewhere. So both BaseApiError and every error type have to force the prototype of it to be kept. It's ugly, it's weird, but it works...

Reply:
https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
Ok, got it. Had no idea of this caveat with transpilers.